internal/oci/uvm_test.go

```diff
@@ -0,0 +1,76 @@
+package oci
+
+import (
+	"testing"
+
+	runhcsopts "github.com/Microsoft/hcsshim/cmd/containerd-shim-runhcs-v1/options"
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func Test_Spec_NO_Update_MemorySize(t *testing.T) {
+
+	opts := &runhcsopts.Options{
+		VmMemorySizeInMb: 3072,
+	}
+	s := &specs.Spec{
+		Linux: &specs.Linux{},
+		Annotations: map[string]string{
+			annotationMemorySizeInMB: "2048",
+		},
+	}
+	updatedSpec := UpdateSpecFromOptions(*s, opts)
+
+	if updatedSpec.Annotations[annotationMemorySizeInMB] == "3072" {
+		t.Fatal("should not have updated annotation to default when annotation is provided in the spec")
+	}
+}
+
+func Test_SpecUpdate_MemorySize(t *testing.T) {
+
+	opts := &runhcsopts.Options{
+		VmMemorySizeInMb: 3072,
+	}
+	s := &specs.Spec{
+		Linux:       &specs.Linux{},
+		Annotations: map[string]string{},
+	}
+	updatedSpec := UpdateSpecFromOptions(*s, opts)
+
+	if updatedSpec.Annotations[annotationMemorySizeInMB] != "3072" {
+		t.Fatal("should have updated annotation to default when annotation is not provided in the spec")
+	}
+}
+
+func Test_Spec_NO_Update_ProcessorCount(t *testing.T) {
```

Comment:
In my head (not saying you have to change anything) I consider golang test names as the growth of changes. So my naming would produce:
Test_SpecUpdate_ProcessorCount_WithAnnotation_NoOpts
Test_SpecUpdate_ProcessorCount_WithAnnotation_WithOpts
Test_SpecUpdate_ProcessorCount_NoAnnotation_NoOpts
Test_SpecUpdate_ProcessorCount_NoAnnotation_WithOpts

Seems to make it easier for me to track what the minute difference is between them. But again you dont have to change this as I know peoples brains are different haha

Reply:
Agree. This is more clear